poc_mobile_verifier/src/speedtests.rs

```diff
@@ -0,0 +1,240 @@
+use crate::{Error, Result};
+use chrono::{DateTime, Duration, NaiveDateTime, Utc};
+use file_store::{speedtest::CellSpeedtest, FileStore};
+use futures::stream::TryStreamExt;
+use helium_crypto::PublicKey;
+use sqlx::{
+    postgres::{types::PgHasArrayType, PgTypeInfo},
+    FromRow, Type,
+};
+use std::{
+    collections::{HashMap, VecDeque},
+    ops::Range,
+};
+
+#[derive(Debug, Clone, Type)]
+#[sqlx(type_name = "speedtest")]
+pub struct Speedtest {
+    pub timestamp: NaiveDateTime,
+    pub upload_speed: i64,
+    pub download_speed: i64,
+    pub latency: i32,
+}
+
+impl PgHasArrayType for Speedtest {
+    fn array_type_info() -> PgTypeInfo {
+        PgTypeInfo::with_name("_speedtest")
+    }
+}
+
+#[derive(FromRow)]
+pub struct SpeedtestRollingAverage {
+    pub id: PublicKey,
```

Comment:
Is the speedtest on the hotspot or on the 5g radio? I’m not sure you can safely key these off public keys

Reply:
speedtest are off the hotspot